### ISSUE TITLE
Flashed image - SSH login "Access Denied"

### ISSUE THREAD

**Issue by cptcl** [billkenney/update_max3_plus3#1]
Hey.. i followed the instructions,.. 
i flashed the image in step 1.
Put back the EMMC and turned printer on.
Display shows "The system starts abnormal"
Wanted to ssh in, but neither MKS or ROOT with passwort makerbase works
"access denied"
![Screenshot 2024-05-20 195447](https://github.com/billkenney/update_max3_plus3/assets/53144288/b29aa0a4-1a34-45da-82d1-56aefe670e84)

need help here :S

**Comment by cptcl** [comment]
Ok.. so login is..
root
pw= 1234
maybe u want to update your guide to this :S

also
What should i use? bash? or zsh?
Am i supposed to create a user names "mks" and set password to "makerbase"?  otherwise the folder  it supposed to be downloadeed "cd /home/mks" would not exist.. right? thats another error i got.

PS: sorry for beeing picky.. u did so much work here and i am very thankful for your guide.

**Comment by billkenney** [comment]
@cptcl i skipped this step, sorry about that. Yes you should create user mks when prompted. I personally use zsh, it has a lot of plugins and is much more customizable. I updated the guide to include these steps.

**Comment by cptcl** [comment]
@billkenney thanks.. i tryed it with bash for now.. try to learn what exactly i have to do...
May i ask u... a few other questions?
Step 3: The x_4.bin .. if its done correctly the file is renamed to X_4.CUR right? other than that i cant see if its done or not.
Step 4: 
```
mks@mkspi:~$ sudo mount /dev/sda1 /mnt ; wget https://github.com/billkenney/update_max3_plus3/raw/main/klipper.uf2 ; mv klipper.uf2 /mnt
[sudo] Passwort für mks:
mount: /mnt: WARNING: source write-protected, mounted read-only.
mount: (hint) your fstab has been modified, but systemd still uses
       the old version; use 'systemctl daemon-reload' to reload.
--2024-05-20 21:18:18--  https://github.com/billkenney/update_max3_plus3/raw/main/klipper.uf2
Auflösen des Hostnamens github.com (github.com)… 140.82.121.4
Verbindungsaufbau zu github.com (github.com)|140.82.121.4|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 302 Found
Platz: https://raw.githubusercontent.com/billkenney/update_max3_plus3/main/klipper.uf2 [folgend]
--2024-05-20 21:18:18--  https://raw.githubusercontent.com/billkenney/update_max3_plus3/main/klipper.uf2
Auflösen des Hostnamens raw.githubusercontent.com (raw.githubusercontent.com)… 2606:50c0:8000::154, 2606:50c0:8001::154, 2606:50c0:8002::154, ...
Verbindungsaufbau zu raw.githubusercontent.com (raw.githubusercontent.com)|2606:50c0:8000::154|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK
Länge: 70144 (68K) [application/octet-stream]
Wird in »klipper.uf2.1« gespeichert.

klipper.uf2.1                                  100%[=================================================================================================>]  68,50K  --.-KB/s    in 0,02s

2024-05-20 21:18:18 (3,38 MB/s) - »klipper.uf2.1« gespeichert [70144/70144]

mv: das Verschieben zwischen Geräten ist fehlgeschlagen: 'klipper.uf2' zu '/mnt/klipper.uf2'; das Ziel kann nicht entfernt werden: Das Dateisystem ist nur lesbar
```

what do i do wrong? Maybe to mention i have a X-Max 3 v2 and my extruder board looks a bit diffrent
![20240520_212420](https://github.com/billkenney/update_max3_plus3/assets/53144288/3e61a524-db3d-4f80-ab0c-641891cecc85)
it says A-4 v1.2

Im stuck at  the moment

**Comment by cptcl** [comment]
sudo mount /dev/sda1 /mnt -w

For write permission then :D
Did it as root,.. File copied

next thing...
Step 7 (after ur update) follow https://github.com/billkenney/max3_plus3_recovery
How do i know if the update runs or even works if i cant use the display in this moment? Display patches comming in step 8

**Comment by billkenney** [comment]
> Step 7 (after ur update) follow https://github.com/billkenney/max3_plus3_recovery
> 
> How do i know if the update runs or even works if i cant use the display in this moment? Display patches comming in step 8

The deb file will install as long as it doesn't give an error. You can run `sudo ls /root` and will see xindi. It should look similar to this: ![image](https://github.com/billkenney/update_max3_plus3/assets/30010560/38c7844c-f007-4491-a2dd-b37ebccad45a)

The screen should turn white for the firmware update and show a progress indicator (see step 11). If it doesn't, check to make sure 800_480.tft is in the root directory. If it's not, you can ssh into your printer, switch to root user `sudo su`, then `cd /root ; wget https://raw.githubusercontent.com/billkenney/max3_plus3_recovery/main/800_480.tft`. Then turn the machine off and on again. 

I think the makerbase services need to be running in order for the screen firmware update to take place.

**Comment by cptcl** [comment]
sudo dpkg -i /home/mks/mksclient-max3.deb 
gave me
/var/lib/dpkg/info/makerbase-client.postinst: Zeile 3: python: Kommando nicht gefunden.
(python command not found)

![Screenshot 2024-05-20 230219](https://github.com/billkenney/update_max3_plus3/assets/53144288/60a765d6-ed08-4155-9847-cb12a03e3068)
but display shows only the system starts abnormal thing

**Comment by billkenney** [comment]
Run steps 9 and 10, then sudo su root, wget the 800_480 file as described in my last post. Looks like the firmware installed for the most part though.

**Comment by cptcl** [comment]
installed sudo apt install python-is-python3
now i get 
  File "/root/merge.py", line 3, in <module>
ModuleNotFoundError: No module named 'ConfigParser'

for Step 10:
```
Failed to enable unit: File makerbase-client.service: Bad message
The unit files have no installation config (WantedBy=, RequiredBy=, Also=,
Alias= settings in the [Install] section, and DefaultInstance= for template
units). This means they are not meant to be enabled using systemctl.

Possible reasons for having this kind of units are:
• A unit may be statically enabled by being symlinked from another unit's
  .wants/ or .requires/ directory.
• A unit's purpose may be to act as a helper for some other unit which has
  a requirement dependency on it.
• A unit may be started when needed via activation (socket, path, timer,
  D-Bus, udev, scripted systemctl call, ...).
• In case of template units, the unit is meant to be enabled with some
  instance name specified.
Failed to start makerbase-client.service: Unit makerbase-client.service failed to load properly, please adjust/correct and reload service manager: Bad message
See system logs and 'systemctl status makerbase-client.service' for details.
Failed to start makerbase-net-mods.service: Unit makerbase-net-mods.service has a bad unit file setting.
See system logs and 'systemctl status makerbase-net-mods.service' for details.
```

Also i had to install KlipperScreen in KIAUH.. is that right? otherwise sudo dpkg -i /home/mks/mksclient-max3.deb will tell 
```
Failed to disable unit: Unit file KlipperScreen.service does not exist.
```

**Comment by billkenney** [comment]
It's probably easier to just create an image of my emmc and upload that... then all you have to do is write that image to your emmc. I'm actually going to work through this again and upload an image that you can write to your emmc. All you'll need to do is install the appropriate printer.cfg file and flash your mcus. Should have it done in the next hour or two.

**Comment by billkenney** [comment]
The address for some of the maker base service files was incorrect. I've corrected them and now they are running.

**Comment by cptcl** [comment]
> It's probably easier to just create an image of my emmc and upload that... then all you have to do is write that image to your emmc. I'm actually going to work through this again and upload an image that you can write to your emmc. All you'll need to do is install the appropriate printer.cfg file and flash your mcus. Should have it done in the next hour or two.

Thank u for everything. I would prefer to go throu your guide and not just copy pasta. Maybe we find some more things that we can adress in your guide. And maybe i can learn a bit more. if everything else fails, a rdy to go image would be great then.

I would start from scratch right now. Using zsh from beginning and follow your guide as tight as possible.

**Comment by billkenney** [comment]
The image is almost done. I tested it and the screen works and I'm able to print. For some reason, I ran into some issues that I didn't run into the first go around. All you have to do is write the image to your emmc, flash the mcus, and install the correct printer.cfg (unless you have the max3 with the bltouch), and run a couple other commands. The image is for an 8gb emmc, and I saw someone post about using the armbian system to expand the image, but I'm not exactly sure how that's done. Would be great if you could figure it out and let me know and I'll add it to the guide.

**Comment by cptcl** [comment]
i rerun the whole process rn and im stuck again with step 5:
```
mkspi:~:# sudo mount /dev/sda1 /mnt -w
mount: /mnt: /dev/sda1 is write-protected but explicit read-write mode requested.
       dmesg(1) may have more information after failed mount system call.
mount: (hint) your fstab has been modified, but systemd still uses
       the old version; use 'systemctl daemon-reload' to reload.
```
and it did not mount at all.. not even read only

**Comment by billkenney** [comment]
okay, can you try restarting with the bottom left button on the back of the extruder pressed until the printer starts up, then post the output of ls /dev/*?
This guide says you need to press both buttons, but Qidi's post said only the bottom left button, and I'm not sure if I had to press both or not. https://github.com/leadustin/QIDI-up2date-english/blob/main/Klipper-Update/update+upgrade.md
I think the address for the X_4.bin file was wrong too. 
If you can get it to mount, try wget https://raw.githubusercontent.com/billkenney/update_max3_plus3/main/X_4.bin ; mv X_4.bin /mnt

**Comment by cptcl** [comment]
```
mks@mkspi:~$  ls /dev/*
/dev/autofs           /dev/gpiochip3  /dev/loop3         /dev/mmcblk1p1    /dev/sda1       /dev/tty10  /dev/tty22  /dev/tty34  /dev/tty46  /dev/tty58  /dev/ttyS3    /dev/vcs4   /dev/vcsu3        /dev/watchdog0
/dev/btrfs-control    /dev/gpiochip4  /dev/loop4         /dev/mmcblk1p2    /dev/sdb        /dev/tty11  /dev/tty23  /dev/tty35  /dev/tty47  /dev/tty59  /dev/ttyS4    /dev/vcs5   /dev/vcsu4        /dev/zero
/dev/cec0             /dev/gpiochip5  /dev/loop5         /dev/mmcblk1rpmb  /dev/sdb1       /dev/tty12  /dev/tty24  /dev/tty36  /dev/tty48  /dev/tty6   /dev/ttyS5    /dev/vcs6   /dev/vcsu5        /dev/zram0
/dev/console          /dev/hwrng      /dev/loop6         /dev/null         /dev/sg0        /dev/tty13  /dev/tty25  /dev/tty37  /dev/tty49  /dev/tty60  /dev/ttyS6    /dev/vcsa   /dev/vcsu6        /dev/zram1
/dev/cpu_dma_latency  /dev/i2c-1      /dev/loop7         /dev/port         /dev/sg1        /dev/tty14  /dev/tty26  /dev/tty38  /dev/tty5   /dev/tty61  /dev/ttyS7    /dev/vcsa1  /dev/vga_arbiter  /dev/zram2
/dev/cuse             /dev/i2c-4      /dev/loop-control  /dev/ppp          /dev/snapshot   /dev/tty15  /dev/tty27  /dev/tty39  /dev/tty50  /dev/tty62  /dev/uhid     /dev/vcsa2  /dev/vhci
/dev/ecryptfs         /dev/initctl    /dev/media0        /dev/psaux        /dev/spidev0.2  /dev/tty16  /dev/tty28  /dev/tty4   /dev/tty51  /dev/tty63  /dev/uinput   /dev/vcsa3  /dev/vhost-net
/dev/fb0              /dev/kmsg       /dev/media1        /dev/ptmx         /dev/stderr     /dev/tty17  /dev/tty29  /dev/tty40  /dev/tty52  /dev/tty7   /dev/uleds    /dev/vcsa4  /dev/video0
/dev/full             /dev/kvm        /dev/media2        /dev/random       /dev/stdin      /dev/tty18  /dev/tty3   /dev/tty41  /dev/tty53  /dev/tty8   /dev/urandom  /dev/vcsa5  /dev/video1
/dev/fuse             /dev/log        /dev/mem           /dev/rfkill       /dev/stdout     /dev/tty19  /dev/tty30  /dev/tty42  /dev/tty54  /dev/tty9   /dev/vcs      /dev/vcsa6  /dev/video2
/dev/gpiochip0        /dev/loop0      /dev/mmcblk1       /dev/rtc          /dev/tty        /dev/tty2   /dev/tty31  /dev/tty43  /dev/tty55  /dev/ttyS0  /dev/vcs1     /dev/vcsu   /dev/video3
/dev/gpiochip1        /dev/loop1      /dev/mmcblk1boot0  /dev/rtc0         /dev/tty0       /dev/tty20  /dev/tty32  /dev/tty44  /dev/tty56  /dev/ttyS1  /dev/vcs2     /dev/vcsu1  /dev/video4
/dev/gpiochip2        /dev/loop2      /dev/mmcblk1boot1  /dev/sda          /dev/tty1       /dev/tty21  /dev/tty33  /dev/tty45  /dev/tty57  /dev/ttyS2  /dev/vcs3     /dev/vcsu2  /dev/watchdog

/dev/block:
179:0  179:1  179:2  179:32  179:64  251:0  251:1  251:2  7:0  7:1  7:2  7:3  7:4  7:5  7:6  7:7  8:0  8:1  8:16  8:17

/dev/bsg:
0:0:0:0  1:0:0:0

/dev/bus:
usb

/dev/char:
10:1    10:183  10:235  116:0   13:64  1:8      189:384  21:1     240:2  252:0  254:5  4:12  4:18  4:23  4:29  4:34  4:4   4:45  4:50  4:56  4:61  4:67  4:8  7:1    7:133  7:6   7:69  81:4
10:124  10:200  10:236  116:16  13:65  189:0    189:385  226:0    243:0  254:0  29:0   4:13  4:19  4:24  4:3   4:35  4:40  4:46  4:51  4:57  4:62  4:68  4:9  7:128  7:134  7:64  7:70  89:1
10:125  10:229  10:237  116:33  1:4    189:1    189:386  226:1    246:0  254:1  4:0    4:14  4:2   4:25  4:30  4:36  4:41  4:47  4:52  4:58  4:63  4:69  5:0  7:129  7:2    7:65  81:0  89:4
10:126  10:231  10:242  1:3     1:5    189:128  189:512  226:128  246:1  254:2  4:1    4:15  4:20  4:26  4:31  4:37  4:42  4:48  4:53  4:59  4:64  4:7   5:1  7:130  7:3    7:66  81:1
10:127  10:232  1:1     13:32   153:0  189:129  1:9      240:0    247:0  254:3  4:10   4:16  4:21  4:27  4:32  4:38  4:43  4:49  4:54  4:6   4:65  4:70  5:2  7:131  7:4    7:67  81:2
10:130  10:234  1:11    13:63   1:7    189:256  21:0     240:1    251:0  254:4  4:11   4:17  4:22  4:28  4:33  4:39  4:44  4:5   4:55  4:60  4:66  4:71  7:0  7:132  7:5    7:68  81:3

/dev/disk:
by-diskseq  by-id  by-label  by-partuuid  by-path  by-uuid

/dev/dri:
by-path  card0  card1  renderD128

/dev/fd:
0  1  2  3

/dev/hugepages:

/dev/input:
by-path  event0  event1  mice  mouse0

/dev/mapper:
control

/dev/mqueue:

/dev/net:
tun

/dev/pts:
0  1  ptmx

/dev/shm:

/dev/snd:
by-path  controlC0  pcmC0D0p  seq  timer

/dev/v4l:
by-id  by-path
mks@mkspi:~$ 
```
What wonders me.. yesterday i did exactly the same and got it to work. But now i have problems here.. crazy

**Comment by billkenney** [comment]
Image has been uploaded. Readme.md has been updated. Try writing that image to your printer. Assuming you've already flashed your mcus, and you have a max3 with a bltouch, you shouldn't have to do anything else. Looks like /dev/sda1 is there though, I'm not sure why you're unable to mount it...

**Comment by cptcl** [comment]
i have probe. no blt.

**Comment by billkenney** [comment]
> i have probe. no blt.

That's fine, you just have to run a couple extra commands. Also, my guess is that you've already flashed all of the mcus. I think you can probably try steps 1, 5-6, 9, and 12 of the updated read me to start. If you get an mcu error, try flashing the mcus again after completing those steps. You should get the update screen after step12, which takes 25-30 min to complete, but your screen should work after that.

**Comment by cptcl** [comment]
> That's fine, you just have to run a couple extra commands. Also, my guess is that you've already flashed all of the mcus. I think you can probably try steps 1, 5-6, 9, and 12 of the updated read me to start. If you get an mcu error, try flashing the mcus again after completing those steps. You should get the update screen after step12, which takes 25-30 min to complete, but your screen should work after that.

Step 5 gave me
```
mks@mkspi:~$ path=$(ls /dev/serial/by-id/*) ; printf "[mcu MKS_THR]\nserial:$path\n" > ~/klipper_config/MKS_THR.cfg ; ln -s ~/klipper_config/MKS_THR.cfg ~/klipper_config/config/MKS_THR.cfg
-bash: /home/mks/klipper_config/MKS_THR.cfg: Datei oder Verzeichnis nicht gefunden
ln: die symbolische Verknüpfung '/home/mks/klipper_config/config/MKS_THR.cfg' konnte nicht angelegt werden: Datei oder Verzeichnis nicht gefunden
mks@mkspi:~$
Message from syslogd@mkspi at May 21 08:08:02 ...
 kernel:[   62.003063] Disabling IRQ #41
```
Folder or file not found
PS: i dont have a folder named klipper_config should i mkdir one?
EDIT: after mkdir klipper_config and mkdir klipper_config/config the step 5 commands runs without error

**Comment by billkenney** [comment]
I just wrote the image to my emmc, and it seems to work fine. The printer boots up, the screen works, etc. I updated step 5 to remove the existing symbolic link, so maybe try it again, but this is what you should see if you run these commands: 
![Screen Shot 2024-05-21 at 1 31 15 AM](https://github.com/billkenney/update_max3_plus3/assets/30010560/1d829f36-2041-416b-84be-94e1c1eba5e8)
I have no idea why you ~/klipper_config wouldn't exist, because it's there on my system and it's the same image.

**Comment by cptcl** [comment]
i tryed to do steps 1,5-6, 9 and 12. Did not work. Display showed same error message.

i now try it with your image and follow the new guide

PS: your image surely ok?
![Screenshot 2024-05-21 083607](https://github.com/billkenney/update_max3_plus3/assets/53144288/a0675e80-ced8-47c6-a9f8-cb42857176e9)

**Comment by billkenney** [comment]
Maybe you downloaded a different image? I downloaded this one: https://github.com/billkenney/update_max3_plus3/releases/download/qidi_update/qidi_update.img.xz, unarchived it, and wrote it to my emmc with balena etcher with no error messages. After booting I ran the commands in the screenshot. If something were wrong with the image, I'm pretty sure I wouldn't be able to boot from it.

**Comment by cptcl** [comment]
i downloaded it from your guide... i now try this link...  and... same error

**Comment by billkenney** [comment]
> i downloaded it from your guide... i now try this link...  and... same error

That's bizarre. I guess I can try to make another image from the one I just flashed...but it takes awhile to compress it.

**Comment by cptcl** [comment]
your old guide where u have to install yourself with kiauh was not hard to follow. Would prefer this way. maybe if u have a bit time we could have a call over discord or so and screen share?

**Comment by billkenney** [comment]
Still there, I just moved it. I've tried to update it along the way: https://github.com/billkenney/update_max3_plus3/blob/main/manual.md

**Comment by billkenney** [comment]
> your old guide where u have to install yourself with kiauh was not hard to follow. Would prefer this way. maybe if u have a bit time we could have a call over discord or so and screen share?

This would work for me. My time zone is CST, so it's almost 2am here. Not sure where you are, but maybe tomorrow afternoon/evening

**Comment by cptcl** [comment]
> > your old guide where u have to install yourself with kiauh was not hard to follow. Would prefer this way. maybe if u have a bit time we could have a call over discord or so and screen share?
> 
> This would work for me. My time zone is CST, so it's almost 2am here. Not sure where you are, but maybe tomorrow afternoon/evening

im gmt +1 Europe i have 9 am in the morning.
My Discord is impact_aut  ID: 179581859133521920 (impact#4859)

Will try to go throu the guide again rn (https://github.com/billkenney/update_max3_plus3/blob/main/manual.md) 
Would need to print something.. so if it doesnt work i would have to revert it for today. Do i need to flash back something on extruder mcu and mcu?

**Comment by billkenney** [comment]
> Would need to print something.. so if it doesnt work i would have to revert it for today. Do i need to flash back something on extruder mcu and mcu?

Yes write the image from here to your emmc: https://wiki.qidi3d.com/en/Memo/flash-emmc. Then follow the steps here to create the mcu flash files and reflash your mcus with the 0.10 klipper software: https://github.com/QIDITECH/QIDI_PLUS3/issues/27#issuecomment-2073932891

I think you probably also need to replace the printer.cfg file with the one for your the max3 with the probe. I'd also recommend keeping an extra emmc or at least a working backup image if you're trying to upgrade.

**Comment by cptcl** [comment]
> > Would need to print something.. so if it doesnt work i would have to revert it for today. Do i need to flash back something on extruder mcu and mcu?
> 
> Yes write the image from here to your emmc: https://wiki.qidi3d.com/en/Memo/flash-emmc. Then follow the steps here to create the mcu flash files and reflash your mcus with the 0.10 klipper software: [QIDITECH/QIDI_PLUS3#27 (comment)](https://github.com/QIDITECH/QIDI_PLUS3/issues/27#issuecomment-2073932891)
> 
> I think you probably also need to replace the printer.cfg file with the one for your the max3 with the probe. I'd also recommend keeping an extra emmc or at least a working backup image if you're trying to upgrade.

i did make a backup of my emmc at the first place. so this should be good. just for the flashing part .. that i need to handle in case

**Comment by billkenney** [comment]
Yeah just write your backup to the emmc then. And follow the steps to flash the mcus, and it should work fine.

**Comment by cptcl** [comment]
btw in your new guide step 3 u have a typo for the github link. Extruder mcu is the same file as step 2 mcu link (x_4.bin)
should be sudo mount /dev/sda1 /mnt -w ; wget https://raw.githubusercontent.com/billkenney/update_max3_plus3/main/klipper.uf2 ; mv klipper.uf2 /mnt

**Comment by cptcl** [comment]
Hm.. intressting...
I restarted the printer and run ```sudo fdisk -1```
and it showed
```
mks@mkspi:~$ sudo fdisk -l
[sudo] Passwort für mks:
Disk /dev/sda: 3,73 MiB, 3915776 bytes, 7648 sectors
Disk model: flash
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x35ac64c4

Device     Boot Start   End Sectors  Size Id Type
/dev/sda1  *      128  7551    7424  3,6M  e W95 FAT16 (LBA)


Disk /dev/mmcblk1: 28,91 GiB, 31037849600 bytes, 60620800 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0xa1e0c477

Device         Boot  Start      End  Sectors  Size Id Type
/dev/mmcblk1p1       32768   557055   524288  256M ea Linux extended boot
/dev/mmcblk1p2      557056 59998208 59441153 28,3G 83 Linux


Disk /dev/mmcblk1boot0: 4 MiB, 4194304 bytes, 8192 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes


Disk /dev/mmcblk1boot1: 4 MiB, 4194304 bytes, 8192 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes


Disk /dev/zram0: 487,3 MiB, 510967808 bytes, 124748 sectors
Units: sectors of 1 * 4096 = 4096 bytes
Sector size (logical/physical): 4096 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes


Disk /dev/zram1: 50 MiB, 52428800 bytes, 12800 sectors
Units: sectors of 1 * 4096 = 4096 bytes
Sector size (logical/physical): 4096 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes
mks@mkspi:~$
Message from syslogd@mkspi at May 21 10:04:22 ...
 kernel:[   56.917081] Disabling IRQ #41
```
So zram1 the last device..
after that i did "hold boot button on extrudder board, push and release reset button and than release boot button"
and rerun ```sudo fdisk -1```
and it showed this
```
mks@mkspi:~$ sudo fdisk -l
Disk /dev/sda: 3,73 MiB, 3915776 bytes, 7648 sectors
Disk model: flash
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x35ac64c4

Device     Boot Start   End Sectors  Size Id Type
/dev/sda1  *      128  7551    7424  3,6M  e W95 FAT16 (LBA)


Disk /dev/mmcblk1: 28,91 GiB, 31037849600 bytes, 60620800 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0xa1e0c477

Device         Boot  Start      End  Sectors  Size Id Type
/dev/mmcblk1p1       32768   557055   524288  256M ea Linux extended boot
/dev/mmcblk1p2      557056 59998208 59441153 28,3G 83 Linux


Disk /dev/mmcblk1boot0: 4 MiB, 4194304 bytes, 8192 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes


Disk /dev/mmcblk1boot1: 4 MiB, 4194304 bytes, 8192 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes


Disk /dev/zram0: 487,3 MiB, 510967808 bytes, 124748 sectors
Units: sectors of 1 * 4096 = 4096 bytes
Sector size (logical/physical): 4096 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes


Disk /dev/zram1: 50 MiB, 52428800 bytes, 12800 sectors
Units: sectors of 1 * 4096 = 4096 bytes
Sector size (logical/physical): 4096 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes


Disk /dev/sdb: 128 MiB, 134217728 bytes, 262144 sectors
Disk model: RP2
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x000a8a8b

Device     Boot Start    End Sectors  Size Id Type
/dev/sdb1           1 262143  262143  128M  e W95 FAT16 (LBA)
mks@mkspi:~$
```
new device called sdb1. Is this the Extruder mcu?
maybe this is what i should use? cause this is mountabe without -w

**Comment by cptcl** [comment]
After using sdb1 instead of sda1 and applying all qidi patches while ignoring all errors i got my update screen. will report if its works after endit printer.cfg for cruise_ratio and get wifi back.

**Comment by cptcl** [comment]
Found another typo ... Old guide step 9 for the probe
```
rm ~/klipper_config/config/printer.cfg ; wget https://raw.githubusercontent.com/billkenney/max3_plus3_recovery/main/printer-max3_probe.cfg ; mkdir -p /home/mks/klipper_config ; mv printer-max3_probe.cfg /home/mks/klipper_config ; ln -s ~/klipper_config/printer.cfg ~/klipper_config/config/printer.cfg
```
but it should be
```
rm ~/klipper_config/config/printer.cfg ; wget https://raw.githubusercontent.com/billkenney/max3_plus3_recovery/main/printer-max3_probe.cfg ; mkdir -p /home/mks/klipper_config ; mv printer-max3_probe.cfg /home/mks/klipper_config/printer.cfg ; ln -s ~/klipper_config/printer.cfg ~/klipper_config/config/printer.cfg
```
mv printer-max3_probe.cfg /home/mks/klipper_config/printer.cfg
not
mv printer-max3_probe.cfg /home/mks/klipper_config

**Comment by cptcl** [comment]
after update... System starts abnormaly. Have to revert back. i hope i have no trouble to flash back

EDIT:  I learned that i have to unplug all USB devices to get my Extruder MCU to be SDA1 .. should also maybe included in your guide.
PS: for make it easier u could merge those 2 files with the qidi patches in one file of your own so someone only had to extract 1 file into the mks folder than switching and looking around what files should be or not.

**Comment by billkenney** [comment]
instructions have been updated, and a new image has been uploaded with some fixes. if you're going to try again, I'd recommend using the image. i wrote it to my emmc with balena etcher and my printer and screen are working perfectly. i'd also recommend that you don't install Qidi's patch files until your printer and screen are working (I'm not using the patch files). so the steps would be 1-7 and 11. assuming you've already updated the screen's firmware, you should be able to skip step 10. if the screen still isn't working, you might try step 10 and see if the progress indicator pops up. balena etcher on my Mac didn't show any errors, but if you get that error again I'd ignore it and continue writing the image. 

As long as the makerbase-client service and Xindi are running, the screen should work. You can check this with `sudo service makerbase-client status`. It should say active and running. The only other reason the screen may not work is if klipper has a startup error.
